### PR TITLE
Make time inputs the same through the UI

### PR DESCRIPTION
**Pull request by bramkragten**
## Proposed change

Introduces `time-input`, support time input in 24 or 12h mode based on user settings.
The old `time-input` is renamed to `duration-input`.

All (I think) old time and duration inputs are replaced with the new ones.

## Type of change

- [ ] Dependency upgrade
- [ ] Bugfix (non-breaking change which fixes an issue)
- [X] New feature (thank you!)
- [ ] Breaking change (fix/feature causing existing functionality to break)
- [X] Code quality improvements to existing code or addition of tests

## Example configuration

```yaml

```

## Additional information

- This PR fixes or closes issue: fixes #
- This PR is related to issue or discussion:
- Link to documentation pull request:

## Checklist

- [ ] The code change is tested and works locally.
- [ ] There is no commented out code in this PR.
- [ ] Tests have been added to verify that the new code works.

If user exposed functionality or configuration variables are added/changed:

- [ ] Documentation added/updated for [www.home-assistant.io][docs-repository]

[docs-repository]: https://github.com/home-assistant/home-assistant.io

### CODE REVIEW

File: src/common/datetime/create_duration_data.ts [hunk 1]
```diff
@@ -0,0 +1,31 @@
...
+  }
+  const { days, minutes, seconds, milliseconds } = duration;
+  let hours = duration.hours || 0;
+  hours = (hours || 0) + (days || 0) * 24;
+  return {
+    hours: hours,
+    minutes: minutes,
+    seconds: seconds,
+    milliseconds: milliseconds,
+  };
```
Comment (ludeeus) [inline src/common/datetime/create_duration_data.ts:30]:
```suggestion
  return {
    hours,
    minutes,
    seconds,
    milliseconds,
  };
```

File: src/components/ha-duration-input.ts [hunk 2]
```diff
@@ -0,0 +1,140 @@
...
+export interface HaDurationData {
+  hours?: number;
+  minutes?: number;
+  seconds?: number;
+  milliseconds?: number;
+}
+
+@customElement("ha-duration-input")
+class HaDurationInput extends LitElement {
+  @property() public data!: HaDurationData;
```
Comment (ludeeus) [inline src/components/ha-duration-input.ts:15]:
```suggestion
  @property({attribute: false}) public data!: HaDurationData;
```

File: src/components/ha-selector/ha-selector-time.ts [hunk 3]
```diff
@@ -19,46 +17,17 @@ export class HaTimeSelector extends LitElement {
...
-        .min=${parts[1]}
-        .sec=${parts[2]}
-        .format=${useAMPM ? 12 : 24}
-        .amPm=${useAMPM && (Number(hours) > 12 ? "PM" : "AM")}
+        .value=${this.value}
+        .locale=${this.hass.locale}
         .disabled=${this.disabled}
-        @change=${this._timeChanged}
-        @am-pm-changed=${this._timeChanged}
         hide-label
```
Comment (ludeeus) [inline src/components/ha-selector/ha-selector-time.ts:26]: Why pass a label if it's hidden?

File: src/dialogs/more-info/controls/more-info-input_datetime.ts [hunk 4]
```diff
@@ -0,0 +1,101 @@
...
+        ${
+          this.stateObj.attributes.has_date
+            ? html`
+                <ha-date-input
+                  .value=${`${this.stateObj.attributes.year}-${this.stateObj.attributes.month}-${this.stateObj.attributes.day}`}
+                  .disabled=${UNAVAILABLE_STATES.includes(this.stateObj.state)}
+                  @value-changed=${this._dateChanged}
+                >
+                </ha-date-input>
+                ${this.stateObj.attributes.has_time ? "," : ""}
```
Comment (ludeeus) [inline src/dialogs/more-info/controls/more-info-input_datetime.ts:31]:
```suggestion
                ${this.stateObj.attributes.has_time ? "," : ""}
```
Is this `,` desired? kinda looks strange, can we do space or some padding instead?

File: src/panels/config/automation/trigger/ha-automation-trigger-row.ts [hunk 5]
```diff
@@ -118,6 +121,20 @@ export default class HaAutomationTriggerRow extends LitElement {
               </mwc-list-item>
             </ha-button-menu>
           </div>
+          ${this._warnings
+            ? html`<div class="warning">
+                ${this.hass.localize("ui.errors.config.editor_not_supported")}:
+                <br />
+                ${this._warnings!.length > 0 && this._warnings![0] !== undefined
```
Comment (ludeeus) [inline src/panels/config/automation/trigger/ha-automation-trigger-row.ts:128]:
```suggestion
                ${this._warnings.length && this._warnings[0] !== undefined
```

File: src/panels/config/automation/trigger/ha-automation-trigger-row.ts [hunk 6]
```diff
@@ -118,6 +121,20 @@ export default class HaAutomationTriggerRow extends LitElement {
...
+                  ? html` <ul>
+                      ${this._warnings!.map(
```
Comment (ludeeus) [inline src/panels/config/automation/trigger/ha-automation-trigger-row.ts:130]:
```suggestion
                      ${this._warnings.map(
```

File: src/components/ha-selector/ha-selector-time.ts (hunk 3 above)
Comment (bramkragten) [reply src/components/ha-selector/ha-selector-time.ts:26]: 🤷‍♂️ Good question...

File: src/dialogs/more-info/controls/more-info-input_datetime.ts (hunk 4 above)
Comment (bramkragten) [reply src/dialogs/more-info/controls/more-info-input_datetime.ts:31]: It was copied from the entity row, but yeah, I kinda agree

**Review by ludeeus — APPROVED** [review]

**Pull request merged**